### DISCUSSION TITLE
Project Proposal Presentation

### DISCUSSION THREAD

**Discussion by Mathanraj-Sharma** [Mathanraj-Sharma/intro-to-ml-sjp#15] [Announcements]
Since we are running out of time, I prefer to have the project presentations online. I will put a schedule based on my availability during coming week. Teams please be prepared to do your presentation via Zoom/GMeet. 

Each team should fill and submit this form during lecture on **13th June 2025**
Project Proposal Template:
[ML_Project_Proposal_Template (1).docx](https://github.com/user-attachments/files/20649863/ML_Project_Proposal_Template.1.docx)

And prepare a presentation deck using similar format for the proposal presentation. 

Follow this thread for future updates regarding proposal presentation.

**Comment by Vidusahan** [comment]
Sir, do we need to create different documents for project proposal and proposal presentation?

**Reply by Mathanraj-Sharma** [reply]
@Vidusahan you need to fill and submit a hard copy of the template I shared above (it is for record keeping). 

And your team need to prepare a presentation deck (which should also contain all the details expected in the template I shared) and present it during your slot.

**Reply by Vidusahan** [reply]
Thank you sir

**Comment by Ajanthanasia** [comment]
Congratulates Bro.

**Comment by Sahan-Madusanka-Perera** [comment]
Sir we can use pre-trained models for this project right?

**Reply by Mathanraj-Sharma** [reply]
@Sahan-Madusanka-Perera, the project aims to evaluate whether you can apply the ML techniques you learn in lectures. Specifically, I wanted to assess whether or not you understand each step of the ML lifecycle and can perform them.

So, please avoid using pre-trained models, unless you plan to experiment with transfer learning for a specific use case.

**Reply by Sahan-Madusanka-Perera** [reply]
Ok sir thank you!

**Comment by Mathanraj-Sharma** [comment]
Team Leads, general instructions for your repo. 

- Create GitHub repo for your project and add your teammates as colloborators.
- The repo name should be concise but meaningful, also related to your project.
- The project should be well structured enough for your ML project, you can use this cookiecutter template to get started https://github.com/Mathanraj-Sharma/mr_ml_boilerplate, or simply consult it and create your own structure.
- Every change to the code base should be created via PR, don't let anyone to push directly to your default branch. Please add branch rules to prevent direct push to default branch. 
- Make sure each member train a models using different algorithms to experiment. You can choose the best performing model from that. 

Take this as an opportunity to experience GitHub best practices. 

Also create a new discussion in this repo with `{Program}_{group_no}_{project name}` and add details of your team members and your github repo link. You can ask your project specific doubts and in that discussion thread. 

I will be tracking each an every project repo to evaluate individual members. Encourage your team members to be active for this project.

**Comment by Mathanraj-Sharma** [comment]
Team Leads, The schedule for the proposal presentations, for Saturday teams (2025/06/21) we will have it before and after lectures. 

For other teams I will send the invitations via GMeet.

| Group No | Team Name                  | Proposal Date & Time | Team Lead Email                   |
|----------|----------------------------|-----------------------|-----------------------------------|
| SE1      | The Bias Busters           | 2025-06-19 09:00 PM   | sudeshhansika@gmail.com           |
| SE2      | The Backprop Brains        | 2025-06-19 09:15 PM   | yehanwic@gmail.com                |
| CS1      | The Overfitters            | 2025-06-20 09:00 PM   | serinipuwakgolla@gmail.com        |
| CS2      | Dropout Legends            | 2025-06-20 09:15 PM   | dunithabeyarathne2021@gmail.com   |
| SE3      | The ReLU Rebels            | 2025-06-19 08:00 PM   | dewniandradi2021@gmail.com        |
| SE4      | The Neural Net-Nuts        | 2025-06-19 08:15 PM   | venumovonara@gmail.com            |
| CS3      | Naive Bayesians            | 2025-06-19 08:30 PM   | amaseww@gmail.com                 |
| CS4      | K-Means Clowns             | 2025-06-19 08:45 PM   | aashinshanaweerakoon@gmail.com    |
| SE5      | The Epoch Enforcers        | 2025-06-20 08:00 PM   | dsk.sandeep.987@gmail.com         |
| SE6      | The Underfit Avengers      | 2025-06-20 08:15 PM   | hsdbandaranayake@gmail.com        |
| CS5      | Support Vibe Machines      | 2025-06-20 08:30 PM   | sahanpereraperera@gmail.com       |
| CS6      | Confusion Matrix Masters   | 2025-06-20 08:45 PM   | ganindudeshapriya@gmail.com       |
| SE7      | The Gradient Geniuses      | 2025-06-21 08:00 AM   | erangakavishanka111@gmail.com     |
| SE8      | The Activation Nation      | 2025-06-21 08:15 AM   | nuhansahk@gmail.com               |
| CS7      | Gradient Descenters        | 2025-06-21 08:30 AM   | rushani824@gmail.com              |
| CS8      | Overengineered Neurons     | 2025-06-21 08:45 AM   | dulith.perera24@gmail.com         |
| CS9      | Hyperparameter Hustlers    | 2025-06-21 11:00 AM   | dulanyas.mus@gmail.com            |

**Reply by Mathanraj-Sharma** [reply]
@Eranga035225 and @dulithperera, please confirm whether team leads' emails are correct

**Reply by VenumiV** [reply]
Sir our team SE 04 Neural Net-Nuts email address is incorrect in the above the correct one is venumivonara@gmail.com
Could you please send the request again Sorry for the inconvenience

**Reply by Mathanraj-Sharma** [reply]
@VenumiV updated the invite, please check

**Comment by kspeiris** [comment]
I wanted to clarify regarding our upcoming project presentation, should all team members present the presentation, or is it sufficient if only the project leader presents on behalf of the group?

**Reply by Mathanraj-Sharma** [reply]
@kavindup20010305 all the team members should be present for the presentation

**Comment by VenumiV** [comment]
Sir, our team couldn’t submit the report last Friday. Do we need to send a soft copy before the presentation, or is it okay if we hand it over during Friday’s lecture?

**Reply by Mathanraj-Sharma** [reply]
@venumiV as I mentioned here, create a discussion thread for your project and attach a softcopy of your proposal there (in pdf format)

https://github.com/Mathanraj-Sharma/intro-to-ml-sjp/discussions/15#discussioncomment-13458884
```
Also create a new discussion in this repo with {Program}_{group_no}_{project name} and add details of your team members and your github repo link. You can ask your project specific doubts and in that discussion thread.
```

**Reply by VenumiV** [reply]
Okay sir

**Comment by Mathanraj-Sharma** [comment]
@Eranga035225 and @dulithperera, ask teams to check their weekly evaluations in their respective discussion thread!
